### DISCUSSION TITLE
DocumentFile.uri doesn't work with AsyncImage ImageRequest.Builder(Context).data(document.uri)

### DISCUSSION THREAD

**Discussion by anonym24** [coil-kt/coil#2055] [Q&A]
DocumentFile.uri doesn't work with AsyncImage ImageRequest.Builder(Context).data(document.uri)

So my app was granted a folder with video files using SAF.

I would like to show preview for each of one in my app:

            AsyncImage(
                model = ImageRequest.Builder(LocalContext.current)
                    .data(documentFile.uri)
                    .crossfade(true)
                    .build(),
                contentDescription = "Media file preview",
                modifier = Modifier
                    .clip(RoundedCornerShape(6.dp))
                    .height(50.dp)
                    .aspectRatio(16f / 9f)
            )
        }

But it doesn't work...

It works with Glide in View System:

      glide.load(documentFile.uri).into(binding.videoFileItemImage)

**Comment by anonym24** [comment]
I tried [GlideImage](https://bumptech.github.io/glide/int/compose.html) beta for compose and it works as well:

            GlideImage(
                model = documentFile.uri,
                transition = CrossFade,
                contentDescription = "Media file preview",
                modifier = Modifier
                    .clip(RoundedCornerShape(6.dp))
                    .height(50.dp)
                    .aspectRatio(16f / 9f)
            )

**Comment by colinrtwhite** [comment]
@anonym24 Which Coil version and what's the exception? Please file an issue for this
